### PR TITLE
feat(editor): Add user activation survey

### PR DESCRIPTION
**Pull request by RicardoE105**
Context in [AD0-415](https://linear.app/n8n/issue/ADO-415/p0-ask-activated-users-for-their-biggest-challenge-modal)

### CODE REVIEW

**Review by madgbec — COMMENTED** [review]
How do I know which cards to activate I have many

**Comment by RicardoE105** [comment]
@madgbec can you please elaborate?

**Comment by madgbec** [comment]
yes I was hoping to get a tutorial on the procees


Sent from Yahoo Mail for iPhone


On Tuesday, March 14, 2023, 12:34, Ricardo Espinoza ***@***.***> wrote:




@madgbec can you please elaborate?

**Review by krynble — CHANGES_REQUESTED** [review]
I'd ask you to centralise all actions that update the user settings in the `user.service.ts` and perhaps, to prevent misuse, read data from db first, merge incoming settings and then save, wdyt? Simply because if a user is logged in on multiple devices or this might cause override when we add more settings in the future.

File: packages/editor-ui/src/components/UserActivationSurveyModal.vue [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+						:label="$locale.baseText('userActivationSurveyModal.form.label')"
+						color="text-dark"
+					>
+						<n8n-input
+							type="textarea"
+							:maxlength="FEEDBACK_MAX_LENGTH"
+							:rows="3"
+							v-model="feedback"
+							@input="onInput"
+							ref="input"
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/UserActivationSurveyModal.vue:42]: Nit: This ref doesn't seem necessary

File: packages/editor-ui/src/components/UserActivationSurveyModal.vue [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+			</div>
+		</template>
+	</Modal>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import Vue from 'vue';
+import Modal from './Modal.vue';
+import { USER_ACTIVATION_SURVEY_MODAL } from '../constants';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/UserActivationSurveyModal.vue:68]: Nit: IMO: We should use aliased import in case we decide to move this file

**Review by krynble — APPROVED** [review]
BE code looks great. Thanks!

**Comment by netroy** [comment]
@RicardoE105 the newly added e2e test is failing on this branch, as well as on master.

**Pull request merged**